### PR TITLE
Documentation for PHP Debugging

### PR DESCRIPTION
**Pull request by kaloyan-raev**
Signed-off-by: Kaloyan Raev <kaloyan.r@zend.com>

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
@bartlomiej-laczkowski Please review the content.

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 1]
```diff
@@ -71,3 +71,116 @@ It is also possible to add this command as a Dockerfile instruction for a custom
...
+## Starting the Zend Debugger Client
+
+Eclipse Che has the Zend Debugger Client integrated in the Web IDE. For launching the Zend Debugger Client:
+
+1. Go to `Run > Edit Debug Configurations` from the main menu.
+1. Create new `PHP` configuration.
+1. Change any settings if necessary. The defaults are usually OK.
+1. Click the `Debug` button.
+
+![php-debug-configuration.png]({{ base }}/docs/assets/imgs/php-debug-configuration.png)
```
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Should change to ``

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 2]
```diff
@@ -71,3 +71,116 @@ It is also possible to add this command as a Dockerfile instruction for a custom
...
+
+1. Set `Debug Host` to `localhost` or `127.0.0.1`.
+1. Set `Debug Port` to `10137`.
+1. Set `Debug Local Copy` to `No`.
+
+Note that it is not the browser that opens the debug session to the Zend Debugger Client. This is done by the PHP engine that runs in the Che workspace. The browser just tells the PHP engine to do so. So the above settings are for the PHP engine (the Zend Debugger module in particular). Thus the `Debug Host` must be set to `localhost` and not the public host of the docker container running the Che workspace.
+
+In the end the zDebug settings should look like this:
+
+![zdebug-settings.png]({{ base }}/docs/assets/imgs/zdebug-settings.png)
```
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Should change to ``

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 3]
```diff
@@ -71,3 +71,116 @@ It is also possible to add this command as a Dockerfile instruction for a custom
...
+After installing it, go to `Extra Stuff > Setttings` to configure the toolbar:
+
+1. Disable the `Debug Local Copy` option.
+1. Switch the `Client/IDE Settings` to `Manual Settings`.
+1. Set `Debug Port` to `10137`.
+1. Set `IP Address` to `127.0.0.1`.
+
+In the end the toolbar settings should look like this:
+
+![zend-debugger-firefox-settings.png]({{ base }}/docs/assets/imgs/zend-debugger-firefox-settings.png)
```
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Should change to ``

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 4]
```diff
@@ -71,3 +71,116 @@ It is also possible to add this command as a Dockerfile instruction for a custom
...
+### Using Z-Ray
+
+[Z-Ray](http://www.zend.com/en/products/server/z-ray) is a productivity tool, part of [Zend Server](http://www.zend.com/en/products/zend_server), that is available in the Zend stack. Z-Ray requires no installation or configuraton. It is injected into the response coming from your PHP app and shown right in the browser you are using for development. 
+
+Among other features, it also has the capability to trigger a debug session:
+
+1. Click on the "bug" button.
+1. Click on `Debug current page`.
+
+![z-ray-debug.png]({{ base }}/docs/assets/imgs/z-ray-debug.png)
```
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Should change to

**Comment by kaloyan-raev** [comment]
@JamesDrummond I changed the image links to the correct format.

**Comment by slemeur** [comment]
@kaloyan-raev Is it approved by @bartlomiej-laczkowski ? I'd like to merge the PR so we can update the docs on Che's website and in the product

**Comment by kaloyan-raev** [comment]
Today is a holiday in Poland, so I doubt @bartlomiej-laczkowski will be able to take a look at it before Monday. @slemeur if you a in a hurry, you can merge it now. We can have another PR for any additional changes.

**Comment by bartlomiej-laczkowski** [comment]
@kaloyan-raev @slemeur 
Indeed, there is a holiday in Poland :) Anyway +1 from me. Looks great, thanks Kaloyan!

**Comment by slemeur** [comment]
Thanks a lot guys :) ! 
Happy holiday !

**Pull request merged**